File: app/src/main/java/co/tpcreative/supersafe/ui/player/PlayerAdapter.java
package co.tpcreative.supersafe.ui.player;
import android.content.Context;
import android.graphics.PorterDuff;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import com.snatik.storage.Storage;
import butterknife.BindView;
import butterknife.OnClick;
import co.tpcreative.supersafe.R;
import co.tpcreative.supersafe.common.adapter.BaseAdapter;
import co.tpcreative.supersafe.common.adapter.BaseHolder;
import co.tpcreative.supersafe.common.entities.ItemEntity;
import co.tpcreative.supersafe.common.services.SuperSafeApplication;
import co.tpcreative.supersafe.common.util.Utils;
import co.tpcreative.supersafe.model.ThemeApp;

public class PlayerAdapter extends BaseAdapter<ItemEntity, BaseHolder> {
    private Context mContext;
    private Storage storage;
    private PlayerAdapter.ItemSelectedListener ls;
    private String TAG = PlayerAdapter.class.getSimpleName();
    private ThemeApp themeApp = ThemeApp.getInstance().getThemeInfo();

    public PlayerAdapter(LayoutInflater inflater, Context context, PlayerAdapter.ItemSelectedListener itemSelectedListener) {
        super(inflater);
        this.mContext = context;
        storage = new Storage(context);
        storage.setEncryptConfiguration(SuperSafeApplication.getInstance().getConfigurationFile());
        ls = itemSelectedListener;
    }
    @Override
    public int getItemViewType(int position) {
        return position;
    }
    @Override
    public int getItemCount() {
        return dataSource.size();
    }
    @Override
    public BaseHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new PlayerAdapter.ItemHolder(inflater.inflate(R.layout.item_player, parent, false));
    }
    public interface ItemSelectedListener {
        void onClickGalleryItem(int position);
    }
    public class ItemHolder extends BaseHolder<ItemEntity> {
        @BindView(R.id.tvTitle)
        TextView tvTitle;
        @BindView(R.id.imgPlaying)
        ImageView imgPlaying;
        private int mPosition;
        private ItemEntity data;
        public ItemHolder(View itemView) {
            super(itemView);
        }
        @Override
        public void bind(ItemEntity mData, int position) {
            super.bind(mData, position);
            this.data = mData;
            if (data.isChecked){
                imgPlaying.setVisibility(View.VISIBLE);
            }
            else{
                imgPlaying.setVisibility(View.INVISIBLE);
            }
            imgPlaying.setColorFilter(SuperSafeApplication.getInstance().getResources().getColor(themeApp.getAccentColor()), PorterDuff.Mode.SRC_IN);
            Utils.Log(TAG,"position :"+ data.isChecked);
            tvTitle.setText(data.title);
            this.mPosition = position;
        }
        @OnClick(R.id.rl_item)
        public void onClickedItem(View view){
            if (ls!=null){
                ls.onClickGalleryItem(mPosition);
            }
        }
    }
}

File: app/src/main/java/co/tpcreative/supersafe/ui/player/PlayerAdapter.kt
package co.tpcreative.supersafe.ui.player
import android.content.Context
import android.graphics.PorterDuff
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatImageView
import androidx.appcompat.widget.AppCompatTextView
import androidx.core.content.ContextCompat
import co.tpcreative.supersafe.R
import co.tpcreative.supersafe.common.adapter.BaseAdapter
import co.tpcreative.supersafe.common.adapter.BaseHolder
import co.tpcreative.supersafe.common.services.SuperSafeApplication
import co.tpcreative.supersafe.common.util.Utils
import co.tpcreative.supersafe.model.ItemModel
import co.tpcreative.supersafe.model.ThemeApp
import kotlinx.android.synthetic.main.activity_player.view.tvTitle
import kotlinx.android.synthetic.main.item_player.view.*

class PlayerAdapter(inflater: LayoutInflater, private val mContext: Context?,private val itemSelectedListener: ItemSelectedListener?) : BaseAdapter<ItemModel, BaseHolder<ItemModel>>(inflater) {
    private val TAG = PlayerAdapter::class.java.simpleName
    private val themeApp: ThemeApp? = ThemeApp.getInstance()?.getThemeInfo()
    override fun getItemViewType(position: Int): Int {
        return position
    }

    override fun getItemCount(): Int {
        return mDataSource.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseHolder<ItemModel> {
        return ItemHolder(inflater!!.inflate(R.layout.item_player, parent, false))
    }

    interface ItemSelectedListener {
        fun onClickGalleryItem(position: Int)
    }

    inner class ItemHolder(itemView: View) : BaseHolder<ItemModel>(itemView) {
        val tvTitle: AppCompatTextView = itemView.tvTitle
        var imgPlaying: AppCompatImageView = itemView.imgPlaying
        private var mPosition = 0
        private var data: ItemModel? = null
        override fun bind(data: ItemModel, position: Int) {
            super.bind(data, position)
            this.data = data
            if (this.data?.isChecked!!) {
                imgPlaying.visibility = View.VISIBLE
            } else {
                imgPlaying.visibility = View.INVISIBLE
            }
            themeApp?.getAccentColor()?.let { ContextCompat.getColor(SuperSafeApplication.getInstance(), it) }?.let { imgPlaying.setColorFilter(it, PorterDuff.Mode.SRC_IN) }
            Utils.Log(TAG, "position :" + this.data?.isChecked)
            tvTitle.text = this.data?.title
            mPosition = position
            itemView.rl_item.setOnClickListener {
                itemSelectedListener?.onClickGalleryItem(mPosition)
            }
        }
    }
}